### ISSUE TITLE
Rule request: redundant set access control

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#1869]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

When declaring a variable, one can add a redundant ACL for setter.

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

This just adds extra noise. I don't think it's common enough for the community actively discuss it.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
class Foo {
  // should trigger
  private(set) private var bar1: Int
  internal(set) internal var bar2: Int
  internal(set) var bar3: Int // I'm not so sure about this one
  fileprivate(set) fileprivate var bar4: Int
  public(set) public var bar5: Int
  open(set) open var bar6: Int

  // shouldn't trigger
  private(set) var bar7: Int
  internal(set) public var bar8: Int
}
```

3. Should the rule be configurable, if so what parameters should be configurable?

No

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Enabled, as I think this is rare enough.

**Comment by jpsim** [comment]
👍 Agreed, this might help find bugs too, since you might forget to remove an explicit setter ACL when changing the ACL of the variable.
